darts/models/forecasting/ensemble_model.py

```diff
@@ -377,6 +382,130 @@ def _predictions_reduction(
             ]
         return predictions[0] if is_single_series else predictions
 
+    def save(
+        self, path: Optional[Union[str, os.PathLike, BinaryIO]] = None, **pkl_kwargs
+    ) -> None:
+        """
+        Saves the ensemble model under a given path or file handle.
+
+        If `forecasting_models` contains TorchForecastingModel, the checkpoint file would be saved.
+
+        Example for saving and loading a :class:`RegressionEnsembleModel`:
+
+            .. highlight:: python
+            .. code-block:: python
+
+                from darts.models import RegressionEnsembleModel, LinearRegressionModel, TiDEModel
+
+                model = RegressionEnsembleModel(
+                    forecasting_models = [
+                        LinearRegressionModel(lags=4),
+                        TiDEModel(input_chunk_length=4, output_chunk_length=4),
+                        ],
+                        regression_train_n_points=10,
+                )
+
+                model.save("my_ensemble_model.pkl")
+                model_loaded = RegressionEnsembleModel.load("my_ensemble_model.pkl")
+            ..
+
+        Parameters
+        ----------
+        path
+            Path or file handle under which to save the ensemble model at its current state. If no path is specified,
+            the ensemble model is automatically saved under ``"{RegressionEnsembleModel}_{YYYY-mm-dd_HH_MM_SS}.pkl"``.
+            If the ith model of `forecasting_models` is a TorchForecastingModel, the checkpoint file would be saved
+            under ``"{path}.{ithModelClass}_{i}.ckpt"``.
+            E.g., ``"RegressionEnsembleModel_2024-07-25_14_58_53.pkl"`` and
+            ``"RegressionEnsembleModel_2024-07-25_14_58_53.pkl.TiDEModel_1.ckpt"``
+        pkl_kwargs
+            Keyword arguments passed to `pickle.dump()`
+        """
+
+        if path is None:
+            # default path
+            path = self._default_save_path() + ".pkl"
+
+        if isinstance(path, (str, os.PathLike)):
+            # save the whole object using pickle
+            with open(path, "wb") as handle:
+                pickle.dump(obj=self, file=handle, **pkl_kwargs)
+        elif isinstance(path, io.BufferedWriter):
+            # save the whole object using pickle
+            pickle.dump(obj=self, file=path, **pkl_kwargs)
+        else:
+            raise_log(
+                ValueError(
+                    "Argument 'path' has to be either 'str' or 'PathLike' (for a filepath) "
+                    f"or 'BufferedWriter' (for an already opened file), but was '{path.__class__}'."
+                ),
+                logger=logger,
+            )
+
+        for i, m in enumerate(self.forecasting_models):
```

Comment:
Is there a reason, why we don't use Darts' built-in save / load functionality for the torch models to handle all of this?

Reply:
Hi @dennisbader, I've checked the `TorchForecastingModel`'s built-in save/load API and found that it creates two files: a `.pt` model object and a `.ckpt` checkpoint file. However, the `.pt` file is not necessary. The structure of the `TorchForecastingModel` is included in the `.pkl` file of the ensemble model, and only needs the `.ckpt` file to recover `LightningModule` checkpoint.

Therefore, to avoid saving the `.pt` file(redundant) while saving the `.ckpt` file, I didn't use the built-in save/load directly. Of course, it would have made the code cleaner if I had used the built-in save/load. I wonder what your opinion is?

Reply:
Yes, it will save the .pt which is technically redundant, but I'd still prefer using the built-in logic. Otherwise, we would have to maintain the torch save/load logic in two places.

Reply:
Indeed, code maintainability should be given prominence, and I'll be using the built-in save and load methods with some modifications to the `save` method comments.